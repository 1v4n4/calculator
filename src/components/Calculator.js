import React, { useState } from 'react';
import '../CSS/calculator.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

const Calculator = () => {
  const [data, setData] = useState({ total: null, next: null, operation: null });

  const handleClick = (name) => {
    setData(Calculate(name, data));
  };

  // const { total, next } = data;

  return (
    <div className="calculatorContainer">
      <h2>Letâ€™s do some math!</h2>
      <div className="calculatorDiv">
        <Display result={data.next || data.total || '0'} />
        <ButtonPanel handleClick={handleClick} />
      </div>
    </div>
  );
};
export default Calculator;
