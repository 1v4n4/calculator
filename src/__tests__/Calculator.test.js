import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders Calculator page correctly', () => {
  const calculator = renderer.create(<Calculator />).toJSON();
  expect(calculator).toMatchSnapshot();
});
