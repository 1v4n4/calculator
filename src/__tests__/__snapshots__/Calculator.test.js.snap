// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator page correctly 1`] = `
<div
  className="calculatorContainer"
>
  <h2>
    Letâ€™s do some math!
  </h2>
  <div
    className="calculatorDiv"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="row"
    >
       
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
         
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
        value="/"
      >
        /
         
      </button>
       
    </div>
    <div
      className="row"
    >
       
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
         
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
        value="*"
      >
        *
         
      </button>
       
    </div>
    <div
      className="row"
    >
       
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
         
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
         
      </button>
       
    </div>
    <div
      className="row"
    >
       
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
         
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
         
      </button>
       
    </div>
    <div
      className="row"
    >
       
      <button
        className="double"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
         
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
         
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
         
      </button>
       
    </div>
  </div>
</div>
`;
